src/entity/player.rs

```diff
@@ -453,9 +453,11 @@ impl CustomEntityRenderer for PlayerRenderer {
                 y_scale: 0.16,
                 x_scale: 0.01,
             };
-            let mut name = format::Component::Text(format::TextComponent::new(&player_model.name));
-            format::convert_legacy(&mut name);
-            state.build(&name, format::Color::Black);
+            let name = format::Component::new(format::ComponentType::new_with_color(
```

Comment:
new_with_color sounds a bit odd, what about with_color or adding an optional to the new function